spec/unit/shell/bash_spec.rb

```diff
@@ -131,7 +131,7 @@ def echo_result
     end
 
     it "sets environment variables if requested" do
-      expect(connection).to receive(:execute).with('FOO=bar sh -c echo\\ \\$FOO')
+      expect(connection).to receive(:execute).with('/usr/bin/env FOO=bar sh -c echo\\ \\$FOO')
```

Comment:
Had to do this because the `FOO=bar command` syntax is not portable (bashism).

But another side effect we may want to take into account is that the jail provider just run a command on the local system and by default the user environment is preserved:

```sh-session
romain@fenchurch ~/Projects/bolt/freebsd % cat inventory.yaml 
targets:
  - puppet
  - puppetdb
  - puppetboard
  - node1
  - node2
config:
  transport: jail
romain@fenchurch ~/Projects/bolt/freebsd % sudo bundle exec bolt command run "puppet agent -t" -t all                                                                                               
Started on puppet...
Started on puppetboard...
Started on node2...
Started on puppetdb...
Started on node1...
Failed on puppet:
  The command failed with exit code 1
  <internal:/usr/local/lib/ruby/site_ruby/3.0/rubygems/core_ext/kernel_require.rb>:85:in `require': cannot load such file -- /usr/local/lib/ruby/gems/3.0/gems/bundler-2.3.23/lib/bundler/setup (LoadError)
  	from <internal:/usr/local/lib/ruby/site_ruby/3.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
Failed on puppetboard:
  The command failed with exit code 1
  <internal:/usr/local/lib/ruby/site_ruby/3.0/rubygems/core_ext/kernel_require.rb>:85:in `require': cannot load such file -- /usr/local/lib/ruby/gems/3.0/gems/bundler-2.3.23/lib/bundler/setup (LoadError)
  	from <internal:/usr/local/lib/ruby/site_ruby/3.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
Failed on node2:
  The command failed with exit code 1
  <internal:/usr/local/lib/ruby/site_ruby/3.0/rubygems/core_ext/kernel_require.rb>:85:in `require': cannot load such file -- /usr/local/lib/ruby/gems/3.0/gems/bundler-2.3.23/lib/bundler/setup (LoadError)
  	from <internal:/usr/local/lib/ruby/site_ruby/3.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
Failed on node1:
  The command failed with exit code 1
  <internal:/usr/local/lib/ruby/site_ruby/3.0/rubygems/core_ext/kernel_require.rb>:85:in `require': cannot load such file -- /usr/local/lib/ruby/gems/3.0/gems/bundler-2.3.23/lib/bundler/setup (LoadError)
  	from <internal:/usr/local/lib/ruby/site_ruby/3.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
Failed on puppetdb:
  The command failed with exit code 1
  <internal:/usr/local/lib/ruby/site_ruby/3.0/rubygems/core_ext/kernel_require.rb>:85:in `require': cannot load such file -- /usr/local/lib/ruby/gems/3.0/gems/bundler-2.3.23/lib/bundler/setup (LoadError)
  	from <internal:/usr/local/lib/ruby/site_ruby/3.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
Failed on 5 targets: puppet,puppetdb,puppetboard,node1,node2
Ran on 5 targets in 0.12 sec
romain@fenchurch ~/Projects/bolt/freebsd % sudo bundle exec bolt command run "env -i USER=root PATH=/sbin:/bin:/usr/sbin:/usr/bin:/usr/local/sbin:/usr/local/bin:/root/bin /usr/local/bin/puppet agent -t" -t all
Started on puppet...
Started on puppetboard...
Started on node2...
Started on puppetdb...
Started on node1...
Finished on node1:
  Info: Using environment 'production'
  Info: Retrieving pluginfacts
  Info: Retrieving plugin
  Info: Loading facts
  Info: Caching catalog for node1.lan
  Info: Applying configuration version 'production-1d15767'
  Notice: Applied catalog in 5.68 seconds
Finished on puppetboard:
  Info: Using environment 'production'
  Info: Retrieving pluginfacts
  Info: Retrieving plugin
  Info: Loading facts
  Info: Caching catalog for puppetboard.lan
  Info: Applying configuration version 'production-1d15767'
  Notice: Applied catalog in 3.42 seconds
Finished on puppetdb:
  Info: Using environment 'production'
  Info: Retrieving pluginfacts
  Info: Retrieving plugin
  Info: Loading facts
  Info: Caching catalog for puppetdb.lan
  Info: Applying configuration version 'production-1d15767'
  Notice: Applied catalog in 2.93 seconds
Finished on node2:
  Info: Using environment 'production'
  Info: Retrieving pluginfacts
  Info: Retrieving plugin
  Info: Loading facts
  Info: Caching catalog for node2.lan
  Info: Applying configuration version 'production-1d15767'
  Notice: Applied catalog in 2.88 seconds
Finished on puppet:
  Info: Using environment 'production'
  Info: Retrieving pluginfacts
  Info: Retrieving plugin
  Info: Loading facts
  Info: Caching catalog for puppet.lan
  Info: Applying configuration version 'production-1d15767'
  Notice: Applied catalog in 3.53 seconds
Successful on 5 targets: puppet,puppetdb,puppetboard,node1,node2
Ran on 5 targets in 17.74 sec
```

Should we sanitize the environment?

```sh-session
romain@fenchurch ~/Projects/bolt/freebsd % sudo bundle exec bolt command run "env" -t puppet
Started on puppet...
Finished on puppet:
  BLOCKSIZE=K
  MM_CHARSET=UTF-8
  MANPATH=/usr/home/romain/.vendor/ruby/3.0/gems/puppet-7.20.0/man:
  BOLT_GEM=true
  BOLT_DISABLE_ANALYTICS=true
  GEM_HOME=/home/romain/.vendor/ruby/3.0
  GEM_PATH=
  RUBYLIB=/usr/local/lib/ruby/gems/3.0/gems/bundler-2.3.23/lib
  RUBYOPT=-r/usr/local/lib/ruby/gems/3.0/gems/bundler-2.3.23/lib/bundler/setup
  PATH=/sbin:/bin:/usr/sbin:/usr/bin:/usr/local/sbin:/usr/local/bin:/root/bin
  BUNDLER_VERSION=2.3.23
  BUNDLE_GEMFILE=/usr/home/romain/Projects/bolt/Gemfile
  BUNDLE_BIN_PATH=/usr/local/lib/ruby/gems/3.0/gems/bundler-2.3.23/exe/bundle
  BUNDLER_ORIG_RUBYOPT=BUNDLER_ENVIRONMENT_PRESERVER_INTENTIONALLY_NIL
  BUNDLER_ORIG_RUBYLIB=BUNDLER_ENVIRONMENT_PRESERVER_INTENTIONALLY_NIL
  BUNDLER_ORIG_RB_USER_INSTALL=BUNDLER_ENVIRONMENT_PRESERVER_INTENTIONALLY_NIL
  BUNDLER_ORIG_PATH=/sbin:/bin:/usr/sbin:/usr/bin:/usr/local/sbin:/usr/local/bin
  BUNDLER_ORIG_MANPATH=BUNDLER_ENVIRONMENT_PRESERVER_INTENTIONALLY_NIL
  BUNDLER_ORIG_GEM_PATH=BUNDLER_ENVIRONMENT_PRESERVER_INTENTIONALLY_NIL
  BUNDLER_ORIG_GEM_HOME=BUNDLER_ENVIRONMENT_PRESERVER_INTENTIONALLY_NIL
  BUNDLER_ORIG_BUNDLER_VERSION=BUNDLER_ENVIRONMENT_PRESERVER_INTENTIONALLY_NIL
  BUNDLER_ORIG_BUNDLE_GEMFILE=BUNDLER_ENVIRONMENT_PRESERVER_INTENTIONALLY_NIL
  BUNDLER_ORIG_BUNDLE_BIN_PATH=BUNDLER_ENVIRONMENT_PRESERVER_INTENTIONALLY_NIL
  LS_COLORS=di=34:ln=35:ex=31:cd=36;43:so=32
  EDITOR=nvim
  TZ=Pacific/Tahiti
  LANG=C.UTF-8
  LC_ALL=fr_FR.UTF-8
  TERM=xterm-256color
  MAIL=/var/mail/root
  LOGNAME=root
  USER=root
  HOME=/root
  SHELL=/bin/csh
  SUDO_COMMAND=/usr/local/bin/bundle exec bolt command run env -t puppet
  SUDO_USER=romain
  SUDO_UID=1001
  SUDO_GID=1001
Successful on 1 target: puppet
Ran on 1 target in 0.01 sec
```

Reply:
As noted by @nicklewis on Slack, `jexec(1)` has a `-l` flag that can be used to take care of this:

```sh-session
romain@fenchurch ~/Projects/bolt/freebsd % sudo bundle exec bolt command run "env" -t puppet
Started on puppet...
Finished on puppet:
  SHELL=/bin/csh
  HOME=/root
  USER=root
  BLOCKSIZE=K
  MAIL=/var/mail/root
  MM_CHARSET=UTF-8
  LANG=C.UTF-8
  PATH=/sbin:/bin:/usr/sbin:/usr/bin:/usr/local/sbin:/usr/local/bin:/root/bin
  TERM=xterm-256color
Successful on 1 target: puppet
Ran on 1 target in 0.01 sec
romain@fenchurch ~/Projects/bolt/freebsd % sudo bundle exec bolt command run 'puppet agent -t' -t all
Started on puppet...
Started on puppetboard...
Started on node2...
Started on puppetdb...
Started on node1...
Finished on node2:
  Info: Using environment 'production'
  Info: Retrieving pluginfacts
  Info: Retrieving plugin
  Info: Loading facts
  Info: Caching catalog for node2.lan
  Info: Applying configuration version 'production-1d15767'
  Notice: Applied catalog in 6.94 seconds
Finished on node1:
  Info: Using environment 'production'
  Info: Retrieving pluginfacts
  Info: Retrieving plugin
  Info: Loading facts
  Info: Caching catalog for node1.lan
  Info: Applying configuration version 'production-1d15767'
  Notice: Applied catalog in 4.27 seconds
Finished on puppetdb:
  Info: Using environment 'production'
  Info: Retrieving pluginfacts
  Info: Retrieving plugin
  Info: Loading facts
  Info: Caching catalog for puppetdb.lan
  Info: Applying configuration version 'production-1d15767'
  Notice: Applied catalog in 3.60 seconds
Finished on puppetboard:
  Info: Using environment 'production'
  Info: Retrieving pluginfacts
  Info: Retrieving plugin
  Info: Loading facts
  Info: Caching catalog for puppetboard.lan
  Info: Applying configuration version 'production-1d15767'
  Notice: Applied catalog in 3.63 seconds
Finished on puppet:
  Info: Using environment 'production'
  Info: Retrieving pluginfacts
  Info: Retrieving plugin
  Info: Loading facts
  Info: Caching catalog for puppet.lan
  Info: Applying configuration version 'production-1d15767'
  Notice: Applied catalog in 3.84 seconds
Successful on 5 targets: puppet,puppetdb,puppetboard,node1,node2
Ran on 5 targets in 17.59 sec
```